### PR TITLE
[nasa/nos3#163] Spacecraft Docker

### PR DESCRIPTION
**Pull request by jlucas9**
Major change to use docker as the default tool to run with. All modules running in docker.

Must close submodule PRs first:
* https://github.com/nasa-itc/generic_radio/pull/5
* https://github.com/nasa-itc/gsw-cosmos/pull/16
* https://github.com/nasa-itc/hwlib/pull/7
* https://github.com/nasa-itc/PSP/pull/3
* https://github.com/nasa-itc/CFS_TO/pull/3
* https://github.com/nasa-itc/to_lab/pull/2
* https://github.com/nasa-itc/deployment/pull/13

### CODE REVIEW

File: components/novatel_oem615
Comment (jlucas9) [inline components/novatel_oem615:1]: Note that this is to include a hotfix on the `dev` branch of novatel_oem615 removing debug prints;

File: fsw/osal
Comment (jlucas9) [inline fsw/osal:1]: Note that this is to include a hotfix on the `nos3-rc` branch of osal reducing the number of debug prints for thread names.

**Comment by jlucas9** [comment]
A quick compare between these changes and lost builds for record:
* Docker - `time make fsw sim` =  real	1m5.121s
* Host - `time make -j fsw sim` = real	0m53.326s

A standard make wasn't used because the latest OpenC3 build actually takes the longest out of the pieces and a host only build wasn't available without it.

**Review by Isgaroth-the-Green — APPROVED** [review]
With the alteration to the docker_build scripts, everything works fine on my VM (and looks reasonable to me).

**Review by msuder — APPROVED** [review]

**Pull request merged**